packages/pluggableWidgets/video-player-native/src/VideoPlayer.tsx

```diff
@@ -36,27 +53,129 @@ export function VideoPlayer(props: VideoPlayerProps<VideoStyle>): ReactElement {
+                    style={styles.fullScreenVideoPlayer}
+                    onBackButtonPress={() => fullScreenHandler(false)}
+                    testID="fullscreen-modal"
+                >
+                    <TouchableWithoutFeedback onPress={onVideoPressHandler} testID="fullscreen-overlay">
+                        <Video
+                            {...videoProps}
+                            ref={fullScreenPlayerRef}
+                            paused={false}
+                            onLoad={data => {
+                                setStatus(StatusEnum.READY);
+                                setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
+                                fullScreenPlayerRef.current?.seek(currentPlayTime);
+                            }}
+                            testID={`${props.name}-fullscreen`}
+                            style={styles.fullScreenVideoStyle}
+                        />
+                    </TouchableWithoutFeedback>
+                    {showControls && (
+                        <TouchableOpacity
+                            style={styles.controlBtnContainerStyle}
+                            onPress={() => {
+                                fullScreenHandler(false);
+                            }}
+                            testID="btn-fullscreen-exit"
+                        >
+                            <Icon name="fullscreen-exit" color="white" size={22} />
+                        </TouchableOpacity>
+                    )}
+                    {status === StatusEnum.LOADING && (
+                        <ActivityIndicator
+                            color={styles.indicator.color}
+                            size="large"
+                            style={styles.fullScreenActivityIndicatorStyle}
+                        />
+                    )}
+                    {errorText}
+                </Modal>
+            )}
+            <View style={styles.container}>
+                {status === StatusEnum.LOADING && <ActivityIndicator color={styles.indicator.color} size="large" />}
+                {errorText}
+                {!fullScreen && (
+                    <TouchableWithoutFeedback style={styles.container} onPress={onVideoPressHandler}>
+                        <Video
+                            {...videoProps}
+                            paused={!props.autoStart && !currentPlayTime}
+                            onLoad={data => {
+                                setStatus(StatusEnum.READY);
+                                setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
+                                playerRef.current?.seek(currentPlayTime);
+                            }}
+                            ref={playerRef}
+                            style={status !== StatusEnum.READY ? { height: 0 } : styles.video}
+                            onProgress={({ currentTime }) => setCurrentPlayTime(currentTime)}
```

Comment:
I suggest using `useCallback`

Reply:
On second thought this may not be helpful. Best to check the component's implementation. Let me know what you think

Reply:
this is the same case as onLoad